src/seahawk_deck/seahawk_deck/motion_controller.py

```diff
@@ -64,19 +72,13 @@ def _callback(self, joy_msg):
         twist_msg.angular.y = controller['right_stick']['y'] # P (pitch) 
         twist_msg.angular.z = controller['right_stick']['x'] # Y (yaw)
 
-        # CONSTANT AXIS SCALING
-        AXIS_SCALE = {
-            'linear':  {'x': 1, 'y': 1, 'z': 1},    # forwards, sideways, depth
-            'angular': {'x': 0, 'y': 0.5, 'z': 0.5},  # roll, pitch, yaw
-        }
-        twist_msg.linear.y  *= AXIS_SCALE['linear']['x']
-        twist_msg.linear.x  *= AXIS_SCALE['linear']['y']
-        twist_msg.linear.z  *= AXIS_SCALE['linear']['z']
-        twist_msg.angular.x *= AXIS_SCALE['angular']['x']
-        twist_msg.angular.y *= AXIS_SCALE['angular']['y']
-        twist_msg.angular.z *= AXIS_SCALE['angular']['z']
-
-        self.twist_pub.publish(twist_msg)
+        # AXIS SCALING
+        twist_msg.linear.x *= self.get_parameter('linear_x_scale').get_parameter_value().double_value
+        twist_msg.linear.y  *= self.get_parameter('linear_y_scale').get_parameter_value().double_value
+        twist_msg.linear.z  *= self.get_parameter('linear_z_scale').get_parameter_value().double_value
+        twist_msg.angular.x *= self.get_parameter('angular_x_scale').get_parameter_value().double_value
+        twist_msg.angular.y *= self.get_parameter('angular_y_scale').get_parameter_value().double_value
+        twist_msg.angular.z *= self.get_parameter('angular_z_scale').get_parameter_value().double_value
```

Comment:
We aren't multiplying the twist in the branch I separated the code. Do you think you could move this down into the motor array generation code?